tornado/iostream.py

```diff
@@ -158,11 +160,14 @@ def __init__(self, io_loop=None, max_buffer_size=None,
                                    self.max_buffer_size // 2)
         self.max_write_buffer_size = max_write_buffer_size
         self.error = None
-        self._read_buffer = collections.deque()
-        self._write_buffer = collections.deque()
+        self._read_buffer = bytearray()
+        self._read_buffer_pos = 0
         self._read_buffer_size = 0
+        self._write_buffer = bytearray()
+        self._write_buffer_pos = 0
         self._write_buffer_size = 0
         self._write_buffer_frozen = False
+        self._pending_writes_while_frozen = []
```

Comment:
I don't think so. When we unfreeze, not all the write buffer has necessarily been written (e.g. on Windows with the 128KB limit).